package-installer/src/main/java/com/vmware/pscoe/iac/installer/Installer.java

```diff
@@ -1380,68 +1385,124 @@ private static List<Package> getFilesystemPackages(final PackageType type) {
+                Files.write(Paths.get(wfErrFilePath), workflowExecutionResult.getError().getBytes(), StandardOpenOption.CREATE);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to perform file operation: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        } catch (JsonSyntaxException e) {
+            throw new RuntimeException("Unable to parse input / output file data: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        } catch (WorkflowExecutionException e) {
+            try {
+                Files.write(Paths.get(wfErrFilePath), e.getMessage().getBytes(), StandardOpenOption.CREATE);
+            } catch (IOException e1) {
+                throw new RuntimeException("Unable to perform file operation: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+            }
+            throw new RuntimeException("Workflow execution failed: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        } catch (Exception e) {
+            throw new RuntimeException("General error during workflow execution: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        }
+    }
+	
+    private static boolean isValidYaml(String yamlString) {
+        Yaml yaml = new Yaml();
+        try {
+            yaml.load(yamlString);
+            return true;
+        } catch (ParserException e) {
+            return false;
+        }
+    }
+
+    private static String convertYamlToJson(String fileName, String yamlString) {
+        Yaml yaml = new Yaml();
+        ObjectMapper jsonMapper = new ObjectMapper();
+        jsonMapper.enable(SerializationFeature.INDENT_OUTPUT);
+
+        try {
+            // Parse YAML
+            Map<String, Object> yamlMap = yaml.load(yamlString);
+            // Convert to JSON
+            return jsonMapper.writeValueAsString(yamlMap);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(String.format("Unable to convert YAML file '%s' to JSON: '%s'", fileName, e.getLocalizedMessage()));
+        }
+    }
+
+    private static String convertJsonObjectToYaml(String fileName, JsonObject jsonObject) {
+        if (jsonObject.isEmpty()) {
+            return "--";
+        }
+        ObjectMapper jsonMapper = new ObjectMapper();
+        DumperOptions options = new DumperOptions();
+        options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+        Yaml yaml = new Yaml(options);
+
+        try {
+            @SuppressWarnings("unchecked")
+            Map<String, Object> jsonMap = jsonMapper.readValue(jsonObject.toString(), Map.class);
+            return yaml.dump(jsonMap);
+        } catch (Exception e) {
+            throw new RuntimeException(String.format("Unable to convert JSON object to YAML file '%s': '%s'", fileName, e.getLocalizedMessage()));
+        }
+    }
+    
+    private static boolean endsWithIgnoreCase(String str, String suffix) {
```

Comment:
Love this, let's use it when validating the input

Reply:
Actually I don't need the exception this method just checks whether this is a valid yaml file.